src/kostentraeger/transformer.ts

```diff
@@ -0,0 +1,142 @@
+        if (!acceptsInternetOrPaperReceipts) {
+            throw new Error(`${messageTxt} Expected that institution at least accepts either paper receipts or receipts sent over internet`)
+        }
+    }
+
+    if (msg.kto) {
+        if (!msg.kto.iban || !msg.kto.bic) {
+            throw new Error(`${messageTxt} Expected IBAN and BIC`)
+        }
+    }
+}
+
+function transformMessage(msg: KTORMessage): Institution | null {
+    /* in practice, this doesn't seem to be used and usage of this field is inconsistent for the
+       different umbrella organizations that issue the Kostenträger-file. "03" however seems to mean
+       "delete this entry" */
+    if (msg.fkt.verarbeitungskennzeichenSchluessel == "03") return null
+
+    return {
+        ik: msg.idk.ik,
+        name: msg.nam.names.join(" "),
+        abbreviatedName: msg.idk.abbreviatedName,
+        
+        bankAccountDetails: msg.kto ? createBankAccountDetails(msg.kto, msg.idk.abbreviatedName) : undefined,
+
+        validityFrom: msg.vdt.validityFrom,
+        validityTo: msg.vdt.validityTo,
+
+        contacts: msg.aspList.map((asp) => createContact(asp)),
+        addresses: msg.ansList.map((ans) => createAddress(ans)),
+        transmissionMethods: createReceiptTransmissionMethods(msg.uemList, msg.dfuList),
+        links: msg.vkgList
+    }
+}
+
+function createReceiptTransmissionMethods(uemList: UEM[], dfuList: DFU[]): ReceiptTransmissionMethods | undefined {
+    if (uemList.length == 0) return undefined
+
+    const machineReadablePaperReceipt = uemList.some((uem) => uem.uebermittlungsmediumSchluessel == "5")
+    const paperReceipt = uemList.some((uem) => uem.uebermittlungsmediumSchluessel == "6")
+
+    let zeichensatzSchluessel, email, ftam
+    if (dfuList.length > 0) {
+        zeichensatzSchluessel = uemList.find((uem) => uem.uebermittlungsmediumSchluessel == "1")?.uebermittlungszeichensatzSchluessel
+        email = dfuList.find((dfu) => dfu.dfuProtokollSchluessel == "070")?.address
+        ftam = dfuList.find((dfu) => dfu.dfuProtokollSchluessel == "016")?.address
+    }
+
+    return {
+        paperReceipt: paperReceipt,
+        machineReadablePaperReceipt: machineReadablePaperReceipt,
+        email: email,
+        ftam: ftam,
+        zeichensatzSchluessel: zeichensatzSchluessel
+    }
+}
+
+function createContact(asp: ASP): Contact {
+    return {
+        phone: asp.phone,
+        fax: asp.fax,
+        name: asp.name,
+        fieldOfWork: asp.fieldOfWork
+    }
+}
+
+function createBankAccountDetails(kto: KTO, abbreviatedName: string): BankAccountDetails {
+    return {
+        bankName: kto.bankName,
+        accountOwner: kto.accountOwner ?? abbreviatedName,
+        iban: kto.iban!,
+        bic: kto.bic!
+    }
+}
+
+function createAddress(ans: ANS): Address {
+    const place = ans.place
+    const postcode = ans.postcode
+    const addr = ans.address
```

Comment:
Object destructuring comes in handy here:
```suggestion
    const { place, postcode, address: addr } = ans
```